working/augmentation-libraries/feature-specification.md

```diff
@@ -1155,9 +1150,11 @@ extensionDeclaration ::=
 extensionBody ::= '{' (metadata classMemberDeclaration)* '}'
 
 extensionTypeDeclaration ::=
-  'augment'? 'extension' 'type' 'const'? typeIdentifier
-  typeParameters? representationDeclaration interfaces?
-  '{' (metadata classMemberDeclaration)* '}'
+    'extension' 'type' 'const'? typeIdentifier
+    typeParameters? representationDeclaration interfaces?
+    '{' (metadata classMemberDeclaration)* '}'
+  | 'augment' 'extension' 'type' typeIdentifier typeParameters? interfaces?
```

Comment:
Are we sure we want to separate the grammar here? This means that if you type:

```dart
augment extension type Foo(int x) {}
```

You will get a parse error. In my experience parse errors lead to extremely bad UX, for example the formatter will choke (on the entire file), and it is likely that everything after this error will give erroneous errors. The error you do get will also likely be something like "unexpected `(`". And not a more specific error saying the representation declaration is not allowed on augmentations of extension types (maybe even with a quick fix to remove it!).

Reply:
That shouldn't matter: The tools have always used a grammar that derives a bigger language than the specified grammar, because that yields a better UX (during recovery and for error messages). Surely they will continue to do so.

Reply:
Ok, if the parser doesn't strictly follow the exact grammar then sgtm